src/pages/index.js

```diff
@@ -171,11 +171,23 @@ class IndexPage extends Component {
 
     async onSearchChange() {
         const { searchValue, showAllClubs } = this.state;
+        const hasSearchValue = searchValue.trim().length > 0;
+        // Preserve results from previous search if current search value is empty
+        if(!hasSearchValue) {
+            return;
+        }
         this.setState({ loading: true });
         this.setParams({ q: searchValue });
         if(!showAllClubs) {
             await this.setPosition(searchValue);
         }
+        else {
+            /*
+             * A hack to get the loading animation to appear. Otherwise, the first update would've
+             * been overwritten by the second update (due to React batching subsequent `setState` calls).
```

Comment:
No idea if this is relevant here, and didn't read this carefully, but check out the component forceUpdate method

Reply:
Tried that and it doesn't work :(

Reply:
Tried `forceUpdate` already but it didn't work